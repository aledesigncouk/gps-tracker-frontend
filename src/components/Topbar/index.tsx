import { JSX } from "react";
import { useStore } from "@store/ContextStore";
import YearSelector from "@components/YearSelector";
import ControlSwitch from "@components/ControlSwitch";
import DateRangeSelector from "@/components/DateRangeSelector";
import { ControlSwitchEnum } from "@enums/enums";

import styles from "@styles/components/Topbar.module.scss";
import "react-datepicker/dist/react-datepicker.css";

const Topbar = (): JSX.Element => {
  const { controlSwitch, setControlSwitch } = useStore();

  return (
    <>
      <div className={styles.topbar}>
        <ControlSwitch onToggle={setControlSwitch} />
        {controlSwitch === ControlSwitchEnum.RANGE ? (
          <DateRangeSelector />
        ) : (
          <YearSelector />
        )}
      </div>
    </>
  );
};

export default Topbar;
