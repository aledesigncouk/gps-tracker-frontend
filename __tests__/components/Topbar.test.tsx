import { render, screen } from '@testing-library/react';
import Topbar from '@components/Topbar';

jest.mock('@components/YearSelector', () => () => <div data-testid="year-selector">Year Selector</div>);
jest.mock('@components/DateRangeSelector', () => () => <div data-testid="date-range-selector">Date Range Selector</div>);

describe('Topbar component', () => {

  beforeEach(() => {
    render(<Topbar />);
  });

  it('renders without crashing', () => {
    const topbar = screen.queryByTestId('topbar');
    expect(topbar).toBeInTheDocument();
  });

  it('renders the DateRangeSelector component', () => {
    const dateRangeSelector = screen.queryByTestId('date-range-selector');
    expect(dateRangeSelector).toBeInTheDocument();
    expect(dateRangeSelector).toHaveTextContent('Date Range Selector');
  });

  it('renders the YearSelector component', () => {
    const yearSelector = screen.queryByTestId('year-selector');
    expect(yearSelector).toBeInTheDocument();
    expect(yearSelector).toHaveTextContent('Year Selector');
  });
});
