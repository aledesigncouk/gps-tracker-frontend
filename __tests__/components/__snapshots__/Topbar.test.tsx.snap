// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar component match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="topbar"
    data-testid="topbar"
  >
    <div
      data-testid="date-range-selector"
    >
      Date Range Selector
    </div>
    <div
      data-testid="year-selector"
    >
      Year Selector
    </div>
  </div>
</DocumentFragment>
`;
